aws/awsutil/strings.go

```diff
@@ -0,0 +1,67 @@
+package awsutil
+
+import (
+	"unicode"
+	"unicode/utf8"
+)
+
+// StringHasPrefixFold tests whether the string s begins with prefix, interpreted as UTF-8 strings,
+// under Unicode case-folding.
+//
+// This implementation is a modified version of the Go Standard Library's strings.EqualFold
+// Copyright 2009 The Go Authors. All rights reserved.
+func StringHasPrefixFold(s, prefix string) bool {
```

Comment:
What do you think about putting this utility in the SDK's `internal/strings`, or `private/protocol/internal/strings`, etc pkg ?  If we decided to remove/change t in the future won't have to worry about breaking changes.